operator/main.py

```diff
@@ -563,50 +559,31 @@ def get_num_pvs(storage):
     try:
         resp = utils_execute(cmd)
         parts = resp.stdout.strip().split("\n")
-        storage.pv_count = int(parts[0].strip())
+        pv_count = int(parts[0].strip())
+        # Handle NoneType error
+        if pv_count is None:
+            return 0
+        return pv_count
 
     except CommandError as msg:
         logging.error(logf(
             "Failed to get size details of the "
-            "storage \"%s\"" % storage.storage_name,
+            "storage \"%s\"" % volname,
             error=msg
         ))
+        # Set default as 0
+        return 0
```

Comment:
When PVC is not provisioned from that volume and is deleted, this block is executed but deletion stops. Hence setting default value as 0.